app/models/project_association.rb

```diff
@@ -37,7 +37,8 @@ class ProjectAssociation < ActiveRecord::Base
   belongs_to :project_b, class_name:  'Project',
                          foreign_key: 'project_b_id'
 
-  validates_presence_of :project_a, :project_b
+  alias_attribute :dependent_project, :project_b
+  validates_presence_of :project_a, :dependent_project
```

Comment:
I think you should either rename the `belongs_to :project_b` directly, if it is always the dependent project (and do the same for `:project_a`) or only rely on I18n to output the human attribute name for the keys `:project_a` and `:project_b`